crates/bls-gadgets/src/hash_to_group.rs

```diff
@@ -243,8 +250,8 @@ impl<P: Bls12Parameters> HashToGroupGadget<P> {
                     return Err(SynthesisError::AssignmentMissing);
                 }
 
-                let x_bits = &xof_bits[..377];
-                let greatest = xof_bits[377];
+                let x_bits = &xof_bits[..X_BITS];
+                let greatest = xof_bits[SIGN_BIT_POSITION];
```

Comment:
```suggestion
                let greatest = xof_bits[377];
```

This is always 377, as we've built the array here: https://github.com/celo-org/bls-zexe/pull/161/files#diff-1c127c26439a36eaf637b12bcfa01e7cR243

Reply:
Good catch!

Reply:
Alternatively, we can forget about building the array and use it directly here